src/elementary/ReactionID.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef NJOY_ELEMENTARY_REACTIONID
+#define NJOY_ELEMENTARY_REACTIONID
+
+// system includes
+#include <string>
+
+// other includes
+#include "elementary/src/split.hpp"
+#include "elementary/Identifier.hpp"
+#include "elementary/ParticlePairID.hpp"
+#include "elementary/ParticleTupleID.hpp"
+
+namespace njoy {
+namespace elementary {
+
+  /**
+   *  @class
+   *  @brief The identifier for a reaction
+   *
+   *  A reaction is represented by an two body entrance reaction channel and
+   *  an n-body exit reaction channel.
+   *
+   *  The symbol of the reaction identifier consists of two valid particle
+   *  tuple identifiers separated by an arrow. For example:
+   *    - "n,Fe56->n,Fe56" for elastic neutron scattering on Fe56
+   *    - "n,he4->h2,h3" for a neutron on alpha resulting in a deuteron and
+   *      triton
+   */
+  class ReactionID : public Identifier< ReactionID > {
+
+    friend Identifier< ReactionID >;
+
+    /* fields */
+
+    /* auxiliary functions */
+    static std::string name() { return "a reaction"; }
+
+  public:
+
+    /* constructor */
+    #include "elementary/ReactionID/src/ctor.hpp"
+
+    /* methods */
+    #include "elementary/ReactionID/src/validate.hpp"
+
+    /**
+     *  @brief Return the incident particle pair identifier for the reaction
+     */
+    ParticlePairID incident() const noexcept {

```

Comment:
incident refers to the incident particle pair in a reaction and outgoing to the outgoing tuple of particles after the reaction. I would not use reactants since that's more of a chemical term than a physical one. I'd prefer these relative neutral names since outgoing could still be a special case like fission, total or a qualifier like [continuum]. In these cases, products has no meaning.